src/hotspot/share/memory/arena.hpp

```diff
@@ -136,7 +134,8 @@ class Arena : public CHeapObj<mtNone> {
   void* operator new(size_t size, const std::nothrow_t& nothrow_constant, MEMFLAGS flags) throw();
   void  operator delete(void* p);
 
-  // Fast allocate in the arena.  Common case is: pointer test + increment.
+  // Fast allocate in the arena.  Common case aligns to the size of long which is 64 bits
+  // on both 32 and 64 bit platforms. Required for atomic long operations on 32 bits.
   void* Amalloc(size_t x, AllocFailType alloc_failmode = AllocFailStrategy::EXIT_OOM) {
     assert(is_power_of_2(ARENA_AMALLOC_ALIGNMENT) , "should be a power of 2");
```

Comment:
[pre-existing] I think this could now be a static_assert.

Reply:
Changing it so.  Why is there capital STATIC_ASSERT and not capital static_assert ?

Reply:
I was sort of underwhelmed with the usefulness of this:
runtime/synchronizer.cpp:        static_assert(is_power_of_2(NINFLATIONLOCKS), "must be");

Reply:
Kind of like the required message for our assert macro.  static_assert was fixed in C++17.